LMR-75 - оживление страницы редактирования инфо об инструменте

Daria-pugacheva opened the pull request:
Убрала лишнюю страницу с фронта и лишние комментарии на странице с инфо об инструменте.

Из профиля убрала кнопки Удалить и Снять с аренды (и их функции)
По странице редактирования инструмента. Пока есть форма редактирования простых полей (категория, название, марка, цена, залог, описание).
По поводу редактирования даты окончания аренды, чтобы снять с аренды, как обсудили на скраме 16.05.22, написала комментарий в Трелло - примем решение по бэку и добавлю тогда (на бэке мне нужно будет подкорректировать ДТО-шку для этого, а Диме логику расчета интервалов доступности)
ПО ПОВОДУ РЕДАКТИРОВАНИЯ ФОТОК. Пока в эту форму не добавляла. Мне видится, что для этого надо отдельно ниже формы со всеми данными как-то выводить имеющиеся фотки с кнопочками "Удалить", чтобы можно было их удалять, и плюс input-поле, чтобы добавлять новые фотки. Т.е. как-то на фронте надо будет редактировать лист фоток (из стартового будут удаляться имеющиеся и добавляться новые фото, а уже результирующий лист pictureUrls отправляться на бэк) Пока в голове не укладывается, как это лучше сделать. Возможно, у Николая, т.к. разбирался с добавлением фоток, есть видение, как это оптимально реализовать? (я посмотрела реализацию отражения фото на странице и добавления, пока все это в голове крутится, но решение, как это объединить для редактирования, пока не нащупывается)
В ревьюеры позвала Николая и Дмитрия, т.к. с них решение по функционалу в странице редактирования. (Дмитрий - корректировка логики интервалов, если решаем, что делаем, Николай - соображения, будем ли делать логику со снятием аренды и может, идеи по редактированию фоток)

Daria-pugacheva committed:
Убрала лишнюю страницу с фронта и лишние комментарии на странице с инфо об инструменте.
Из профиля убрала кнопки Удалить и Снять с аренды (и их функции)
По странице редактирования инструмента. Пока есть форма редактирования простых полей (категория, название, марка, цена, залог, описание).
По поводу редактирования даты окончания аренды,чтобы снять с аренды, как обсудили на скраме 16.05.22, написала комментарий в Трелло - примем решение по бэку и добавлю тогда (на бэке мне нужно будет подкорректировать ДТО-шку для этого, а Диме логику расчета интервалов доступности)
ПО поводу редактирования фоток. Пока в эту форму не добавляла. Мне видится, что для этого надо отдельно ниже формы со всеми данными как-то выводить имеющиеся фотки с кнопочками "Удалить", чтобы можно было их удалять и плюс input-поле, чтобы добавлять новые фотки. Т.е. как-то на фронте надо будет редактировать лист фоток (из стартового будут удаляться и добавляться фото) Пока в голове не укладывается, как это лучше сделать. Возможно, у Николая, т.к. разбирался с добавлением фоток, есть видение,как это оптимально реализовать?

Daria-pugacheva committed:
В форме добавлено поле для изменения endDate (нужно это поле добавить в DTO)

siporqueno reviewed (commented):
По endDate ответил в карточке Trello. По изменениям фотографий - как обсудили выносим в отдельную задачу, чтобы не задерживать эту.

Daria-pugacheva committed:
Merge remote-tracking branch 'origin/develop' into LMR-75

Daria-pugacheva committed:
Merge branch 'develop' into LMR-75

# Conflicts:
#	source/core/src/main/java/ru/letmerent/core/controllers/InstrumentController.java

siporqueno commented on source/core/src/main/resources/static/edit-tool/edit-tool.html:
```diff
@@ -1,23 +1,28 @@
         <p></p>
         <input class="form-control" type="text" placeholder="Описание" ng-model="updated_tool.description">
         <p></p>
+        <p>Если вы хотите снять инструмент с аренды, поменяйте дату окончания предоставления инструмента в аренду на текущую:</p>
```
```suggestion
        <p>Если вы хотите ограничить по времени дальнейшую аренду инструмента, вы можете поменять дату окончания предоставления инструмента в аренду на любую дату после даты окончания крайней аренды: {{earliestPossibleEndDate}}</p>
```

Daria-pugacheva committed:
Починила в profile.js логику, чтобы при попытке попасть  ЛК неавторизованного полльзователя на бэк не летели некорректные запросы (с пустыми данными) и чуть поправила внешний вид личных данных профиля.

Daria-pugacheva committed:
Update source/core/src/main/resources/static/edit-tool/edit-tool.html

Согласна. Когда подолью правки Алексея, еще поменяю название переменной, видимо, и, конечно, в метод проверочку добавлю

Co-authored-by: Nikolay Balashov <46228837+siporqueno@users.noreply.github.com>

Daria-pugacheva committed:
Update source/core/src/main/resources/static/edit-tool/edit-tool.html

Согласна. Когда подолью правки Алексея, еще поменяю название переменной, видимо, и, конечно, в метод проверочку добавлю

Co-authored-by: Nikolay Balashov <46228837+siporqueno@users.noreply.github.com>

Daria-pugacheva committed:
Merge branch 'develop' into LMR-75

Daria-pugacheva committed:
Merge branch 'LMR-75' of https://github.com/siporqueno/let-me-rent into LMR-75

 Conflicts:
	source/core/src/main/java/ru/letmerent/core/controllers/InstrumentController.java

Daria-pugacheva committed:
Добавила проверку, чтобы дата окончания аренды в форме изменения документа не была меньше даты окончания последней аренды.
Пока наброски. Работоспособность еще не проверяла. Посмотрю ближе к 16 часам сегодня

Daria-pugacheva committed:
Добавила проверку, чтобы дата окончания аренды в форме изменения документа не была меньше даты окончания последней аренды.
Пока наброски. Работоспособность еще не проверяла. Посмотрю ближе к 16 часам сегодня

Daria-pugacheva committed:
Merge branch 'develop' into LMR-75

Daria-pugacheva committed:
Merge remote-tracking branch 'origin/develop' into LMR-profileFix

Daria-pugacheva committed:
В html поменяла тип данных в поле ввода даты, потому что ангуляр ругается, что если такой тип, то модель должна быть Date объектом. Пока не разобралась, как это сделать, заменила на text.
В IntervalDto добавила аннотацию NoArgsConstructor, потому что был бубнеж про то, что нет возможности сериализовать.
А дальше провалилась в проблему, которая когда-то всплывала на корзине: Jackson не поддерживает сериализацию LocalDateTime  и , соответственно, не справляется с такими полями как intervals и всеми датами.

Daria-pugacheva committed:
В html поменяла тип данных в поле ввода даты, потому что ангуляр ругается, что если такой тип, то модель должна быть Date объектом. Пока не разобралась, как это сделать, заменила на text.
В IntervalDto добавила аннотацию NoArgsConstructor, потому что был бубнеж про то, что нет возможности сериализовать.
А дальше провалилась в проблему, которая когда-то всплывала на корзине: Jackson не поддерживает сериализацию LocalDateTime  и , соответственно, не справляется с такими полями как intervals и всеми датами.

Daria-pugacheva commented:
Проверка работоспособности вывела меня на сложности с десериализацией из-за наличия в InstrumentInfoDto полей, которые являются LоcalDateTime. Написала в коммите, что  добавила  в код. Принт-скрин с ошибкой прикрепила к карточке в Трелло.  Пока успела только увидеть, какие сыплются ошибки. 
В ревьюеры добавила Алексея, т.к. он помогал с доп.полями в InstrumentInfoDto,  и Артема, т.к. он, если не ошибаюсь, участвовал в решении аналогичной проблемы с корзиной. 
Я сильно не уверена, что оперативно сумею разобраться с тем, как решить эту проблему с сериализацией.

Haazaad commented on source/core/src/main/java/ru/letmerent/core/dto/IntervalDto.java:
```diff
@@ -1,12 +1,7 @@
-import lombok.Getter;
-import lombok.Setter;
-import lombok.ToString;
+import lombok.*;
```
Избавиться от wildcard, необходимо проводить импорт только того, что необходимо для работы

Daria-pugacheva committed:
В странице отзывов об арендаторе добавлена проверка на заполненность полей и сделан выпадающий список для рейтинга

siporqueno committed:
Added InstrumentInfoDtoDeserializer (to be corrected, is empty) and put it to InstrumentInfoDto via annotation.

siporqueno committed:
Added InstrumentInfoDtoDeserializer (to be corrected, is empty) and put it to InstrumentInfoDto via annotation.

Daria-pugacheva committed:
В ветке по изменению инструмента у меня пока сложности возникли (я изначально там эту правку сделала, но ветка пока подвисла). Поэтому я отдельно хочу просто удалить лишние кнопки из профиля, чтобы оперативно они уже в девелопе не маячили и не мешали записывать видео по другим блокам функционала

siporqueno committed:
Added InstrumentInfoDeserializer to solve the issue with deserialization.

siporqueno committed:
Added InstrumentInfoDeserializer to solve the issue with deserialization.

siporqueno reviewed (commented):
Добавил InstrumentInfoDtoDeserializer и аннотацию в InstrumentInfoDto.

Ошибки с десериализацией ушли. Инструмент удается изменить, но он не изменяется а задваивается, то есть сохраняется как новый.

Во-вторых, в конвертере никак не учитываются полученные с фронта интервалы и урлы фоток. Интервалы просто должны остаться как были, а сейчас в новом инструменте их нет. А вот по урлам фоток важно чтобы у измененного инструмента они были как прилетели с фронта. Это понадобится при добавлении возможности удалять/добавлять фото.

siporqueno committed:
Added deserialization of intervals and pictureUrls.

siporqueno committed:
Added deserialization of intervals and pictureUrls.

Commit:
Добавление Minio

Commit:
Добавление Minio

Haazaad commented on source/core/src/main/java/ru/letmerent/core/dto/InstrumentInfoDto.java:
```diff
@@ -1,17 +1,17 @@
-import lombok.Getter;
-import lombok.Setter;
-import lombok.ToString;
+import lombok.*;
```
Избавится от wildcard

Haazaad commented on source/core/src/main/java/ru/letmerent/core/dto/InstrumentInfoDtoDeserializer.java:
```diff
@@ -0,0 +1,113 @@
+
+        JsonNode node = jp.getCodec().readTree(jp);
+        Long id = node.get("id").longValue();
+        System.out.println(id);
```
А зачем это в консоль выводить?

siporqueno replied:
> А зачем это в консоль выводить?

Когда писал десериализатор, смотрел что он выдает. Много лишних кавычек выловил из-за которых бэк инструмент не менял. Убрал все sout.

Haazaad commented on source/core/src/main/java/ru/letmerent/core/dto/InstrumentInfoDtoDeserializer.java:
```diff
@@ -0,0 +1,113 @@
+import java.util.Iterator;
+import java.util.List;
+
+public class InstrumentInfoDtoDeserializer extends StdDeserializer<InstrumentInfoDto> {
```
А смысл этого десериализатора? Неужели ObjectMapper не подхватает?

siporqueno replied:
Не подхватывает, закомментируй @JsonDeserialize(using = InstrumentInfoDtoDeserializer.class) в InstrumentInfoDto и будут ошибки. В карточке задачи есть StackTrace.

siporqueno committed:
Removed sout and unnecessary imports.

siporqueno committed:
Removed sout and unnecessary imports.

siporqueno committed:
Merge pull request #106 from siporqueno/develop

Пролив develop

siporqueno commented:
Не надо пока мержить эту ветку. InstrumentInfoDtoDeserializer срабатывает, естественно, и при добавлении нового инструмента и его надо будет править, чтобы он правильно срабатывал при добавлении. Сейчас срабатывает не правильно и добавление не работает с ним.

siporqueno committed:
Amended InstrumentInfoDtoDeserializer for the case of adding new Instrument.

siporqueno committed:
Amended InstrumentInfoDtoDeserializer for the case of adding new Instrument.

Commit:
правки по загрузке фото

siporqueno committed:
Merge branch 'develop' into LMR-75

# Conflicts:
#	source/core/src/main/resources/static/edit-tool/edit-tool.html
#	source/core/src/main/resources/static/profile/profile.html

Daria-pugacheva committed:
Merge branch 'develop' into Fix-deleteButtonsFromProfile

Daria-pugacheva committed:
Merge pull request #101 from siporqueno/Fix-deleteButtonsFromProfile

Удаление кнопок Удалить и Снять с аренды из профиля.

Daria-pugacheva committed:
Merge pull request #100 from siporqueno/FIx-feedbackAboutRenter

В отзывах об арендаторе добавлен выпадающий список

Daria-pugacheva committed:
Merge branch 'develop' into LMR-profileFix

Daria-pugacheva committed:
Merge pull request #94 from siporqueno/LMR-profileFix

Починила в profile.js логику, чтобы не было пустых запросов на бэк

Commit:
правки фронта

Commit:
правки фронта

Commit:
правки фронта

siporqueno committed:
Передвинул js файлы для страниц вниз

Передвинул js файлы для страниц вниз перед закрывающим тэгом </body>

siporqueno committed:
Передвинул js файлы для страниц вниз

Передвинул js файлы для страниц вниз перед закрывающим тэгом </body>

siporqueno committed:
Realized upload of several pictures by one request.

siporqueno committed:
Resolved conflicts in InstrumentConverter.java during rebase.

siporqueno committed:
Resolved conflicts between local minio-2 and origin minio-2

siporqueno committed:
Merge pull request #108 from siporqueno/minio-2

Minio-2 feature

siporqueno committed:
Removed import lombok.* and added exact necessary imports for lombok.

siporqueno committed:
Removed import lombok.* and added exact necessary imports for lombok.

siporqueno committed:
Conflict resolution during rebase

siporqueno committed:
Resolved conflicts during rebase

siporqueno committed:
Resolved conflicts with origin
